MetaMask/metamask-mobile
# feat: add bridge wallet action

## **Description**



### Motivation
Organically exposing MetaMask Mobile users to MetaMask Bridges could contribute to a company-wide initiative to close the gap between our current and targeted revenue projections. Mobile users currently only contribute [~5% of MetaMask Portfolio transactions](https://mixpanel.com/s/3zHPun) despite [mobile users making up over 62% of page views](https://mixpanel.com/s/1sR9lu); there is an opportunity to capture a vast and wildly underrepresented market here.

### Solution
Add a MetaMask Bridges **wallet** action

## **Related issues**

[METABRIDGE-459](https://consensyssoftware.atlassian.net/browse/METABRIDGE-459)

## **Manual testing steps**

1. Load MetaMask Mobile and set-up/create an account.
2. Selected a supported MetaMask Bridges chain (e.g., Ethereum Mainnet)
3. Tap on the wallet actions button on the bottom-middle of the home screen
   - Verify that the "Bridge" row is present with a matching icon and description
4. Tap on the "Bridge" row
   - Verify that MetaMask Bridges opens in the browser on the selected chain.
5. Go back to the home screen
6. Repeat steps 1-4
   - Verify that the existing tab is reopened in the browser.
7. Switch to an unsupported MetaMask Bridges chain
8. Verify that MetaMask Bridges is no longer an available wallet action

## **Screenshots/Recordings**



### **Before**


<img width="453" alt="Screenshot 2023-11-21 at 5 45 10 PM" src="https://github.com/MetaMask/metamask-mobile/assets/44588480/4bafb1e2-de79-4c72-a0c3-4e7887aa140e">


### **After**


<img width="453" alt="Screenshot 2023-11-21 at 5 46 50 PM" src="https://github.com/MetaMask/metamask-mobile/assets/44588480/893385db-6aac-442b-a79d-714b761df200">

#### iPhone SE

<img width="302" alt="Screenshot 2023-11-29 at 5 38 04 PM" src="https://github.com/MetaMask/metamask-mobile/assets/44588480/9a28ca25-0744-439b-94bf-740837382f19">


#### Happy path flows
https://github.com/MetaMask/metamask-mobile/assets/44588480/24a1d24e-5ce7-43ec-a32f-067150d16800

#### Button does not show on unsupported bridge networks

https://github.com/MetaMask/metamask-mobile/assets/44588480/23f3b712-6585-4107-9c2a-017edb36b0f1


#### Opening bridge with MetaMask Portfolio, Uniswap, and a non-web3 tab open

https://github.com/MetaMask/metamask-mobile/assets/44588480/3ff9b3af-fe1d-4fcf-a099-cb990a117dc0


#### Restarting the app with the same tabs open

https://github.com/MetaMask/metamask-mobile/assets/44588480/a69caa4b-08cd-427f-8c7b-940c23da7fd2


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
